lib/SIL/SILOwnershipVerifier.cpp

```diff
@@ -645,6 +645,14 @@ OwnershipCompatibilityUseChecker::visitAllocRefInst(AllocRefInst *I) {
           UseLifetimeConstraint::MustBeLive};
 }
 
+OwnershipUseCheckerResult
+OwnershipCompatibilityUseChecker::visitReallocRefInst(ReallocRefInst *I) {
+  assert(I->getNumOperands() != 0
+         && "If we reach this point, we must have a tail operand");
+  return {compatibleWithOwnership(ValueOwnershipKind::Trivial),
```

Comment:
You need to handle the passed in owned value here. You should consume it. I would do something like:

```
if (getValue() == I->getReallocPointer()) {
  return {compatibleWithOwnership(ValueOwnershipKind::Owned),
               UseLifetimeConstraint::MustBeInvalidated};
}
assert(getValue() == I->getReallocSize());
  return {compatibleWithOwnership(ValueOwnershipKind::Trivial),
               UseLifetimeConstraint::MustBeLive};
```